### PR TITLE
Added basic PPoly and CubicHermiteSpline interpolation

### PR DESCRIPTION
**Pull request by chrisflesher**
A very basic implementation. I wanted the cubic spline to implement a particular loss function.

### CODE REVIEW

**Comment by jakevdp** [comment]
Hi - thanks for the pull request!

We're in the middle of a process of better defining what's in and out of scope for `jax.numpy` and `jax.scipy`; with the current proposal in https://github.com/google/jax/pull/18137 I think this contribution would have to be considered out of scope.

That document is still in draft form – feel free to weigh in if you have thoughts!

**Comment by chrisflesher** [comment]
Ok, sounds like this is not going to make it.
